packages/syft/src/syft/service/user/user_service.py

```diff
@@ -84,6 +91,207 @@ def create(
+                linked_obj=link,
+            )
+
+            method = context.server.get_service_method(NotificationService.send)
+            result = method(context=root_context, notification=message)
+
+            message = CreateNotification(
+                subject="User requested password reset.",
+                from_user_verify_key=user.verify_key,
+                to_user_verify_key=root_key,
+                linked_obj=link,
+            )
+
+            result = method(context=root_context, notification=message)
+            if isinstance(result, SyftError):
+                return result
+
+            return SyftSuccess(message=success_msg)
+
+        # Email notification is Enabled
+        # Therefore, we can directly send a message to the user with its new password.
+        message = CreateNotification(
+            subject="You requested a password reset.",
+            from_user_verify_key=root_key,
+            to_user_verify_key=user.verify_key,
+            linked_obj=link,
+            notifier_types=[NOTIFIERS.EMAIL],
+            email_template=PasswordResetTemplate,
+        )
+
+        method = context.server.get_service_method(NotificationService.send)
+        result = method(context=root_context, notification=message)
+        if isinstance(result, SyftError):
+            return result
+
+        return SyftSuccess(message=success_msg)
+
+    @service_method(
+        path="user.request_password_reset",
+        name="request_password_reset",
+        roles=ADMIN_ROLE_LEVEL,
+    )
+    def request_password_reset(
+        self, context: AuthedServiceContext, uid: UID
+    ) -> str | SyftError:
+        result = self.stash.get_by_uid(credentials=context.credentials, uid=uid)
+        if result.is_err():
+            return SyftError(
+                message=(
+                    f"Failed to retrieve user with UID: {uid}. Error: {str(result.err())}"
+                )
+            )
+        user = result.ok()
+        if user is None:
+            return SyftError(message=f"No user exists for given: {uid}")
+
+        user_role = self.get_role_for_credentials(user.verify_key)
+        if user_role == ServiceRole.ADMIN:
+            return SyftError(
+                message="You can't request password reset for an Admin user."
+            )
+
+        user.reset_token = self.generate_new_password_reset_token(
+            context.server.settings.pwd_token_config
+        )
+        user.reset_token_date = datetime.now()
+
+        result = self.stash.update(
+            credentials=context.credentials, user=user, has_permission=True
+        )
+        if result.is_err():
+            return SyftError(
+                message=(
+                    f"Failed to update user with UID: {uid}. Error: {str(result.err())}"
+                )
+            )
+
+        return user.reset_token
+
+    @service_method(
+        path="user.reset_password", name="reset_password", roles=GUEST_ROLE_LEVEL
```

Comment:
Does the User have some method to generate a `token`? Otherwise they might not be able to use this endpoint right?

Reply:
Ah, I see, the flow makes more sense now. Thanks!